File: spoj/02-srednie/ROZ.cc
#include <bits/stdc++.h>
using namespace std;

int main()
{
    cin.tie(NULL);
    ios_base::sync_with_stdio(0);
    
    int houses, a, b, c, x, y, z, row1, row2, row3;
    
    cin >> houses;
    cin >> row1 >> row2 >> row3;

    while(--houses)
    {
        cin >> a >> b >> c;
        
        x = a + min(row2, row3);
        y = b + min(row1, row3);
        z = c + min(row1, row2);
        
        row1 = x;
        row2 = y;
        row3 = z;
}
    
    cout << min(row1, min(row2, row3)) << "\n";    
    return 0;
}

File: spoj/02-srednie/ROZ.rb
row1, row2, row3 = 0, 0, 0

gets.to_i.times do
    a, b, c = gets.split(' ').map(&:to_i)
       
    tmp1 = a + [row2, row3].min
    tmp2 = b + [row1, row3].min
    tmp3 = c + [row1, row2].min
    
    row1, row2, row3 = tmp1, tmp2, tmp3
end

puts [row1, row2, row3].min